File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.py
class Solution(object):
    def minSubArrayLen(self, target, nums):
        """
        :type target: int
        :type nums: List[int]
        :rtype: int
        """
        n = len(nums)
        left = 0
        currSum = 0
        minLen = sys.maxsize
        
        for i in range(0, n):
            currSum += nums[i]
            
            while(currSum >= target):
                length = i - left + 1
                minLen = min(length, minLen)
                currSum -= nums[left]
                left += 1
                
        if(minLen == sys.maxsize): return 0
        return minLen

File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int n = nums.length;
        int currSum =0;
        int left = 0;
        int minLen = Integer.MAX_VALUE;
        for(int i=0; i<n; i++){
            currSum += nums[i];
            
            while(currSum >= target){
                minLen = Math.min(minLen, i-left+1);
                currSum -= nums[left];
                left++;
            }
        }
        if(minLen == Integer.MAX_VALUE) return 0;
        return minLen;
    }
}